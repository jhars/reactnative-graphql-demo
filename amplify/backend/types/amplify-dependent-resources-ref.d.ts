export type AmplifyDependentResourcesAttributes = {
  "auth": {
    "laxappdemo451d064d": {
      "AppClientID": "string",
      "AppClientIDWeb": "string",
      "IdentityPoolId": "string",
      "IdentityPoolName": "string",
      "UserPoolArn": "string",
      "UserPoolId": "string",
      "UserPoolName": "string"
    }
  }
}