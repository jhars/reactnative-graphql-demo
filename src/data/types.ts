export type ById<T> = { [index: string]: T };

export interface DropPlayerRequestObject {
  leagueId: number;
  playerId: number;
  rosterSpot: RosterSpot;
  rosterId: string;
  playerInfo: Player;
}

export interface SelectDropPlayerObject {
  leagueId: number;
  playerId: number;
  rosterSpot: RosterSpot;
  rosterId: string;
  playerInfo: Player;
}

export enum SortColumnName {
  Name = "Name",
  Points = "Points",
  Position = "Position",
}

export enum Position {
  G = "G",
  D = "D",
  LSM = "LSM",
  SSDM = "SSDM",
  FO = "FO",
  M = "M",
  A = "A",
}

export enum RosterSpot {
  GOALIE = "GOALIE",
  DEFENSE1 = "DEFENSE1",
  DEFENSE2 = "DEFENSE2",
  LSM = "LSM",
  SSDM = "SSDM",
  FO = "FO",
  MIDFIELD1 = "MIDFIELD1",
  MIDFIELD2 = "MIDFIELD2",
  ATTACK1 = "ATTACK1",
  ATTACK2 = "ATTACK2",
  BENCH1 = "BENCH1",
  BENCH2 = "BENCH2",
  BENCH3 = "BENCH3",
  BENCH4 = "BENCH4",
}

interface StatLine {
  points: number;
  goals: number;
  assists: number;
  turnovers: number;
  groundBalls: number;
  faceoffsWon: number;
  saves: number;
}

interface Statistics {
  statLineLastSeason: StatLine;
}

export interface Player {
  id: string;
  firstName: string;
  lastName: string;
  statistics: Statistics;
  position: Position;
}

export interface PlayerData {
  players: Player[];
}

export interface Players {
  data: PlayerData;
}

export interface League {
  id: string;
  title: string;
  teams: Team[];
}

export interface LeaguesData {
  leagues: League[];
}

// interface Leagues {
//   data: LeaguesData;
// }

export interface Team {
  id: string;
  name: string;
  league: League;
}

export interface TeamsData {
  teams: Team[];
}

// interface Teams {
//   data: TeamsData;
// }

export interface Roster {
  id: string;
  teamId: number;
  teamInfo: Team;
  goalie: Player;
  defense1: Player;
  defense2: Player;
  lsm: Player;
  ssdm: Player;
  fo: Player;
  midfield1: Player;
  midfield2: Player;
  attack1: Player;
  attack2: Player;
}

export interface RosterData {
  roster: Roster;
}

interface AddPlayerToTeamRosterData {
  addPlayerToTeamRoster: Roster;
}

export interface AddPlayerToTeamRosterMutationResponse {
  data: AddPlayerToTeamRosterData;
}

//============================
//============================
export interface CurrentUser {
  id: string;
  email: string;
  preferred_username: string;
}
