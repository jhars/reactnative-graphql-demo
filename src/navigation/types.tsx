import { RouteProp, NavigatorScreenParams, ParamListBase, CompositeScreenProps, CompositeNavigationProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import {DrawerNavigationProp} from '@react-navigation/drawer';
import { Team, Position, RosterSpot, League } from '../data/types';
import type { NativeStackScreenProps, NativeStackNavigationProp } from '@react-navigation/native-stack';

//JH-NOTE: review all of this... necessary? 
export type BaseNavigationProps = NativeStackNavigationProp<ParamListBase>;

export type AvailablePlayersScreenRouteProp = RouteProp<MyTeamsStackParamList, 'AvailablePlayersScreen'>;

export type MyTeamsScreenNavigationProps = NavigatorScreenParams<MyTeamsStackParamList>;

export type LeagueTeamsScreenRouteProps = RouteProp<LeaguesStackParamList, 'LeagueTeamsScreen'>;
export type LeaguesScreenNavigationProps = NavigatorScreenParams<LeaguesStackParamList>;

export type CreateNewTeamConfirmationScreenRouteProp = RouteProp<CreateTeamStackParamList, 'CreateNewTeamConfirmation'>;
export type CreateNewTeamNavigationProps = NavigatorScreenParams<CreateTeamStackParamList>;

export type RosterScreenRouteProps = RouteProp<LeaguesStackParamList|MyTeamsStackParamList,'RosterScreen'>;
// export type RosterScreenRouteProps = RouteProp<ParamListBase>;
// export type RosterScreenRouteProps = NavigatorScreenParams<RosterScreenParamList>;
// export type RosterScreenRouteProps = RouteProp<ParamListBase>;
// export type RosterScreenRouteProps = NativeStackNavigationProp<RosterScreenParamList>;

// export type RosterScreenCompositeProps = CompositeNavigationProp<
//   NativeStackNavigationProp<ParamListBase, T>,
//   RouteProp<RosterScreenParamList>
// >;

// export type RootStackScreenProps<T extends keyof RootDrawerParamList> = StackScreenProps<RootDrawerParamList, T>;


// export type RosterScreenCompositeProps<T extends keyof RosterScreenParamList> =
//   CompositeScreenProps<
//     RouteProp<ParamListBase, T>,
//     RootStackScreenProps<keyof RootStackParamList>
//   >;



export type RootDrawerParamList = {
  Home: undefined;
  MyTeams: MyTeamsScreenNavigationProps;
  Leagues: LeaguesScreenNavigationProps;
  Players: undefined;
  Account: undefined;
  CreateTeam: CreateNewTeamNavigationProps;
};

export type CreateTeamStackParamList = { 
  SelectLeagueFromList: undefined,
  CreateNewTeamConfirmation: {
    leagueID: number;
    leagueTitle: string;
  }
}

export type RosterScreenParamList = {
  team: Team | undefined;
  myTeam: boolean | undefined;
}

export type MyTeamsStackParamList = {
  UserTeams: undefined,
  RosterScreen: RosterScreenParamList,
  AvailablePlayersScreen: { 
    availableForLeagueId: number | undefined;
    team: Team | undefined;
    position: Position | undefined;
    rosterId: string | undefined;
    rosterSpot: RosterSpot | undefined;
  }
};

export type LeaguesStackParamList = {
  LeaguesScreen: undefined,
  LeagueTeamsScreen: {
    league: League;
  },
  RosterScreen: RosterScreenParamList
};



declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootDrawerParamList {}
  }
}
