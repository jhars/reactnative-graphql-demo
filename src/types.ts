export type ById<T> = { [index: string]: T };

interface StatLine {
  points: number;
  goals: number;
  assists: number;
  turnovers: number;
  groundBalls: number;
  faceoffsWon: number;
  saves: number;
}

interface Statistics {
  statLineLastSeason: StatLine;
}

interface Player {
  id: string;
  firstName: string;
  lastName: string;
  jersey: number;
  statistics: Statistics;
}

interface PlayerData {
  players: Player[];
}

export interface Players {
  data: PlayerData;
}

interface League {
  id: serial;
  title: string;
  teams: Team[];
}

interface LeagueData {
  leagues: League[];
}

interface Leagues {
  data: LeagueData;
}

export interface Team {
  id: string;
  name: string;
}

interface TeamData {
  teams: Team[]
}

export interface Teams {
  data: TeamData
}

export interface Roster {
  teamId: number;
  goalie: Player;
  defense1: Player;
  defense2: Player;
  lsm: Player;
  fo: Player;
  midfield1: Player;
  midfield2: Player;
  attack1: Player;
  attack2: Player;
}

interface RosterData {
  roster: Roster;
}

export interface Roster {
  data: RosterData;
}

//============================
//============================
export interface CurrentUser {
  id: serial;
  email: string;
}