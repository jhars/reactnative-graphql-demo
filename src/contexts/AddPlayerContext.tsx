import React, { createContext, useState } from 'react';

export const AddPlayerContext = createContext();

export const AddPlayerProvider = ({ children }) => {
  const [addPlayer, setAddPlayer] = useState(null);
  // const [show, setShow] = useState(null);

  return (
    <AddPlayerContext.Provider value={{ addPlayer, setAddPlayer }}>
      {children}
    </AddPlayerContext.Provider>
  );
};