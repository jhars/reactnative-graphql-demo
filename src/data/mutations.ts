import { gql } from '@apollo/client';

export const ADD_TEAM_TO_LEAGUE = gql`
	mutation AddTeam($name: String!, $leagueId: Int!, $ownerId: ID!) {
	  addTeam(name: $name, leagueId: $leagueId, ownerId: $ownerId) {
	    name
	    leagueId
	    ownerId
	    league {
	      title
	      id
	    }
	  }
	}
`;
export const ADD_PLAYER_TO_TEAM = gql`
mutation AddPlayerToTeamRoster($leagueId: Int!, $teamId: Int!, $playerId: Int!, $position: Position!, $rosterSpot: RosterSpot!,$rosterId: ID!) {
  addPlayerToTeamRoster(leagueId: $leagueId, teamId: $teamId, playerId: $playerId, position: $position, rosterSpot: $rosterSpot, rosterId: $rosterId) {

      goalie {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      defense1 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }
    
      defense2 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      lsm {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      fo {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      ssdm {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      midfield1 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      midfield2 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      attack1 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      attack2 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      } 
    }

}
`;