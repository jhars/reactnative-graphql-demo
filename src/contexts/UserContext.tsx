import React, { ReactNode, createContext, useState } from "react";
import { CurrentUser } from "../data/types";

const emptyCurrentUser: CurrentUser = {
  id: "",
  email: "",
  preferred_username: "",
};

interface Props {
  children?: ReactNode;
  // any props that come into the component
}

export const UserContext = createContext({
  user: emptyCurrentUser,
  setUser: (user: CurrentUser) => {},
});
export const UserProvider = ({ children }: Props) => {
  const [user, setUser] = useState(emptyCurrentUser);

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};
