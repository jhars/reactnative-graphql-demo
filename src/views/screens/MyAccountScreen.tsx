import React, {useContext} from 'react';
import { View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import { useAuthenticator } from '@aws-amplify/ui-react-native';
import { UserContext } from "../../contexts/UserContext";
import { ButtonStyles, ContainerStyles } from '../styles/index';

export default function MyAccountScreen() {
  const { signOut } = useAuthenticator();
  const { user } = useContext(UserContext);

  return (
    <View style={ContainerStyles.container}>
      <View style={styles.userData}>
        <Text style={styles.userRow}>Email: {user?.email}</Text>
        <Text style={styles.userRow}>Username: {user?.preferred_username}</Text>
      </View>
      <View style={styles.footer}>
        <TouchableOpacity style={ButtonStyles.actionButton} onPress={signOut}>
          <Text style={ButtonStyles.actionButtonText}>Sign Out</Text>
        </TouchableOpacity>
      </View>

    </View>
  );
}

const styles = StyleSheet.create({
  userData: {
    flex: 1,
    marginTop: 25
  },
  userRow: {
    margin: 10,
    fontSize: 20,
    color: 'white',
  },
  footer: {
    height: 100,  
    justifyContent: 'flex-end',
    flexGrow: 1,
    paddingBottom: 50
  },
})


