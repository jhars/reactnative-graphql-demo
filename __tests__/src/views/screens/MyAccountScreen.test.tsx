/**
 * @format
 */

import 'react-native';
import React from 'react';
import MyAccountScreen from '../../../../src/views/screens/MyAccountScreen';
import { it, describe } from '@jest/globals';
import { render, screen, waitFor } from '@testing-library/react-native';
import { CurrentUser } from '../../../../src/data/types';
import { MOCK_CURRENT_USER } from '../../../mocks/MOCKS_SHARED'
import { MockedProvider } from '@apollo/react-testing';
import { UserContext } from '../../../../src/contexts/UserContext';

const mockCurrentUser: CurrentUser = MOCK_CURRENT_USER
const mockCurrentUserID = MOCK_CURRENT_USER.id
const mockEmail = MOCK_CURRENT_USER.email
const mockPrefUserName = MOCK_CURRENT_USER.preferred_username

jest.mock('@aws-amplify/ui-react-native', () => {
  return {
    useAuthenticator: () => ({
      signOut: jest.fn(),
    }),
  };
});

describe('My Account Screen Tests', () => {

	it('renders with current user email and username', async () => {
	  const MyAccount = render(
	  	<UserContext.Provider value={{user: mockCurrentUser}}>
			  	<MockedProvider addTypename={false}>
			  			<MyAccountScreen />
			  	</MockedProvider>
		  	</UserContext.Provider>
	  )

	  const testedTree = MyAccount.toJSON()
	  
    await waitFor(() => {
      expect(screen.getByText(`Email: ${mockEmail}`)).toBeVisible()
      expect(screen.getByText(`Username: ${mockPrefUserName}`)).toBeVisible()
      expect(testedTree).toMatchSnapshot()
    });
	});

});


