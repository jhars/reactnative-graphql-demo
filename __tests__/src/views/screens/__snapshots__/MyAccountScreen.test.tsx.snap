// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Account Screen Tests renders with current user email and username 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(6 6 48 / 1.0)",
      "flex": 1,
      "padding": 5,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "marginTop": 25,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "margin": 10,
        }
      }
    >
      Email: 
      current@User.edu
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "margin": 10,
        }
      }
    >
      Username: 
      CurrentUser
    </Text>
  </View>
  <View
    style={
      {
        "flexGrow": 1,
        "height": 100,
        "justifyContent": "flex-end",
        "paddingBottom": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "midnightblue",
          "borderRadius": 20,
          "height": 35,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "aliceblue",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        }
      >
        Sign Out
      </Text>
    </View>
  </View>
</View>
`;
